src/sentry/static/sentry/app/views/performance/realUserMonitoring/measurementsHistogramQuery.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {Location} from 'history';
+import omit from 'lodash/omit';
+
+import {Client} from 'app/api';
+import EventView, {isAPIPayloadSimilar} from 'app/utils/discover/eventView';
+import withApi from 'app/utils/withApi';
+
+import {HistogramData} from './types';
+
+type RawHistogramData = {
+  key: string;
+  bin: number;
+  count: number;
+};
+
+type ChildrenProps = {
+  isLoading: boolean;
+  error: string | null;
+  histograms: Record<string, HistogramData[]>;
+};
+
+type Props = {
+  api: Client;
+  location: Location;
+  orgSlug: string;
+  eventView: EventView;
+  // these measurement names should NOT be prefixed with `measurements.`
+  measurements: string[];
+  numBuckets: number;
+  children: (props: ChildrenProps) => React.ReactNode;
+  min?: string;
+  max?: string;
+};
+
+type State = {
+  fetchId: symbol | null;
+} & ChildrenProps;
+
+class MeasurementsHistogramQuery extends React.Component<Props, State> {
+  state: State = {
+    isLoading: true,
+    fetchId: null,
+    error: null,
+    histograms: {},
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    // Reload data if we aren't already loading,
+    const refetchCondition = !this.state.isLoading && this.shouldRefetchData(prevProps);
```

Comment:
doesn't blocking while loading mean that updates to the ux while it's loading throw away the api calls it needs to make to be correct for the ux? not sure if that is a pattern we use but ideally we'd cancel inflight requests and re-issue the latest (with throttling)

Reply:
:+1: to cancelling inflight requests if possible. Just be careful that your API instance isn't shared as clearing a client's queue cancels all requests.

Reply:
Addressed in 320cb5a